### DISCUSSION TITLE
The database is missing some indexes.

### DISCUSSION THREAD

**Discussion by burnclouds** [nextcloud/all-in-one#2959] [Questions] [answered]
The overview for my new instance shows the following warning:

There are some warnings regarding your setup.
The database is missing some indexes. Due to the fact that adding indexes on big tables could take some time they were not added automatically. By running "occ db:add-missing-indices" those missing indexes could be added manually while the instance keeps running. Once the indexes are added queries to those tables are usually much faster.
Missing index "systag_by_tagid" in table "oc_systemtag_object_mapping".

I ran:
```
sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ db:add-missing-indices
Check indices of the share table.
Check indices of the filecache table.
Check indices of the twofactor_providers table.
Check indices of the login_flow_v2 table.
Check indices of the whats_new table.
Check indices of the cards table.
Check indices of the cards_properties table.
Check indices of the calendarobjects_props table.
Check indices of the schedulingobjects table.
Check indices of the oc_properties table.
Check indices of the oc_jobs table.
Check indices of the oc_direct_edit table.
Check indices of the oc_preferences table.
Check indices of the oc_mounts table.
Check indices of the oc_systemtag_object_mapping table.
Done.
```

Warning remains. Suggestions appreciated.

**Comment by szaimen** [comment, marked as answer]
This will be resolved with Nextcloud 27.0.1 stable.

**Comment by heyvaldemar** [comment]
hey everyone!

Ran into those pesky "missing index" warnings in Nextcloud running in Docker? Here’s a quick fix:

**1. Find Your Container:** First up, let's figure out which container is running Nextcloud:
```bash
docker ps
```

**2. Run the Command:** Replace `nextcloud-container-name` with your container's name. If your error message points to a different user ID than `33`, use that one:
```bash
docker exec -u 33 -it nextcloud-container-name php occ db:add-missing-indices
```

**3. Check if It Worked:** Confirm everything's sorted by running:
```bash
docker exec -u 33 -it nextcloud-container-name php occ status
```

**Pro Tip:** This might take a while for large databases, so maybe run it when your server's not too busy. Oh, and always back up your data before you poke around!

If you're setting up for the first time or need more detailed steps on getting Nextcloud up and running with Docker Compose, check out my guide: [How to Install Nextcloud Using Docker Compose](https://www.heyvaldemar.com/install-nextcloud-using-docker-compose/).

Happy tweaking!
